.teamcity/src/main/kotlin/model/GradleSubprojectProvider.kt

```diff
@@ -36,9 +36,8 @@ data class JsonBasedGradleSubprojectProvider(private val jsonFile: File) : Gradl
     override val subprojects = JSON.parseArray(jsonFile.readText()).map { toSubproject(it as Map<String, Any>) }
     private val nameToSubproject = subprojects.map { it.name to it }.toMap()
 
-    override fun getSubprojectsFor(testConfig: TestCoverage, stage: Stage) =
-        subprojects.filter { it.hasTestsOf(testConfig.testType) }
-            .filterNot { testConfig.os.ignoredSubprojects.contains(it.name) }
+    override fun getSubprojectsForFunctionalTest(testConfig: TestCoverage) =
+        subprojects.filter { it.hasTestsOf(testConfig.testType) && it.name != "docs" /* We handle docs specially. see `DocsTest.kt` */ }
```

Comment:
❓ There are more than `docsTest` in that subproject, right? For example the unit tests should still keep running as part of quick feedback. How does this work now?

Reply:
This change actually reveals long-time flaw in our build: we run `quickTest` in QuickFeedback and `platformTest` in PullRequestFeedback but both `quickTest` and `platformTest` include unit tests. There is no way to separate unit tests from them, we had to use `-x test` before.

Thus, there is no (easy) way to say "only run `docs:test` in QuickFeedback, but `docs:platformTest -x docs:test` in DocsTest. The intention of this change was running `docs:test` as part of `DocTest` in PullRequestFeedback. `docs:test` only has 4 unit cases that don't change frequently so I think it's fine. 

If you feel running `docs:test` in PullRequestFeedback is too late, we'd better fix the root cause, for example create a new 'All Unit Tests', like this:

```
+-- QuickFeedbackLinux
      +-- SanityCheck
      +-- AllUnitTestLinux
      +-- QuickFeedbackLinuxJava18
            +--- ....
+-- QuickFeedback
      +-- AllUnitTestWindows
      +-- QuickFeedbackWIndowsJava8
            +--- ....
```

Then stop making `platformTest/quickTest` depend on `test`.

Reply:
We don't need an all unit test build. This is already taken care by the bucket split, right? It seems like `docsTest` should be a different test type and `quickTest` or `platformTest` shouldn't depend on it, right? Then we could treat the `docs` project as a regular project with only unit tests. That means basically turn your proposition around and instead of treading the unit tests of the docs project different, we'd treat the integration tests/docs test of the project different. WDYT?

Reply:
So the dependency between `platformTest` and `docsTest` was done by `platformTest` -> [`checkSamples` -> `docsTest`](https://github.com/gradle/guides/blob/3c80c172770e4309bea0cd0f94c7af2e5d61bc0d/subprojects/gradle-guides-plugin/src/main/java/org/gradle/docs/samples/internal/SamplesDocumentationPlugin.java#L341), which means we have been executing `checkSamples` in every `platformTest` bucket... 

This is another reason we should move `docs` out of normal functional tests.

I removed the dependency between `platformTest` and `checkSamples`, so now `docs:platformTest` only runs unit tests, making `docs` more like a subproject with unit tests only.

In `DocsTest`, `checkSamples` is only run in the last bucket to avoid duplication.